### PR TITLE
Experimental TSDB: Add support for local filesystem backend

### PR DESCRIPTION
**Pull request by slim-bean**
**What this PR does**:

Add support for TSDB blocks to be saved to the local filesystem via the implementation which already exists for this in Thanos.

Used #2083 as a template

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

Signed-off-by: Edward Welch <edward.welch@grafana.com>

### CODE REVIEW

File: pkg/storage/tsdb/backend/filesystem/bucket_client.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package filesystem
+
+import (
+	"github.com/go-kit/kit/log"
+	"github.com/thanos-io/thanos/pkg/objstore"
+	"github.com/thanos-io/thanos/pkg/objstore/filesystem"
+)
+
+// NewBucketClient creates a new filesystem bucket client
+func NewBucketClient(cfg Config, name string, logger log.Logger) (objstore.Bucket, error) {
```
Comment (pstibrany) [inline pkg/storage/tsdb/backend/filesystem/bucket_client.go]: I would leave out unused parameters completely. Or name them `_` to indicate that they are unused.

**Comment by pstibrany** [comment]
> What's the use case for having this in Cortex?

As Marco points out, one can simply disable shipping to the external storage, which is effectively the same.

**Comment by pracucci** [comment]
I talked to Ed offline. Ed's use case is running a single-node Cortex offline and then having a periodic rsync to synch blocks to secondary Cortex cluster. Despite this can be done disabling the shipper and having an rsync of blocks (not WAL), from a UX perspective looks more cleaner using the `filesystem` storage backend and having an `rsync` with the filesystem `dir` as source. Moreover the shipper manipulates the `meta.json` (ie. adds the external labels) and looks the safest approach in this case.

**Comment by pracucci** [comment]
> I think the use case is for single node cortex, because cortex can run on arm arch and not prometheus.

My take is that for a single node Cortex with blocks storage you don't need the `filesystem` support. You can just run Cortex single binary disabling the shipper.

**Comment by cyriltovena** [comment]
Yeah I don't have enough knowledge on this !

**Pull request merged**